cronjobs/cronjob.go

```diff
@@ -0,0 +1,108 @@
+package cronjobs
+
+import (
+	"log"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// NewCronManager creates a new CronManager.
+func NewCronManager() *CronManager {
+	return &CronManager{
+		stop: make(chan struct{}),
+	}
+}
+
+// AddJob adds a new job to the cron manager.
+func (c *CronManager) AddJob(spec string, task func()) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	job := &Job{
+		Spec: spec,
+		Task: task,
+	}
+
+	c.jobs = append(c.jobs, job)
+}
+
+// Start starts the cron manager.
+func (c *CronManager) Start() {
+	for _, job := range c.jobs {
+		go func(job *Job) {
+			c.runJob(job)
+		}(job)
+	}
+}
+
+// Stop stops the cron manager.
+func (c *CronManager) Stop() {
+	close(c.stop)
+}
+
+// runJob executes the task associated with a given Job at scheduled intervals.
+// It continuously calculates the next run time based on the Job's cron expression,
+// sleeps until that time, and then triggers the associated task.
+func (c *CronManager) runJob(job *Job) {
+	for {
+		select {
+		case <-c.stop:
+			return
+		default:
+			now := time.Now()
+			nextRun := parseCronExpression(job.Spec, now)
+
+			sleepDuration := time.Until(nextRun)
+
+			select {
+			case <-c.stop:
+				return
+			case <-time.After(sleepDuration):
+				job.Task()
+			}
+		}
+	}
+}
+
+// parseCronExpression parses a cron expression and calculates the next run time.
+func parseCronExpression(spec string, now time.Time) time.Time {
```

Comment:
Why are we taking the default times as current time?
It will raise bugs rights, instead can we through an error log to the user